TCAT/Controllers/RouteDetailDrawerViewController+Extensions.swift

```diff
@@ -40,6 +41,65 @@ extension RouteDetailDrawerViewController: LargeDetailTableViewDelegate {
 
 }
 
+extension RouteDetailDrawerViewController: NotificationToggleTableViewDelegate {
+
+    func displayNotificationBanner(type: NotificationType) {
+        guard let direction = getFirstDirection() else { return }
+        FloatingNotificationBanner(
+            customView: NotificationBannerView(
+                busAttachment: getBusIconImageAsTextAttachment(for: direction.routeNumber),
+                type: type
+            )
+        ).show(
+            queue: NotificationBannerQueue(maxBannersOnScreenSimultaneously: 1),
+            on: navigationController
+        )
+    }
+
+    private func getFirstDirection() -> Direction? {
+        return route.directions.first(where: { $0.type == .depart })
+    }
+
+    private func getBusIconImageAsTextAttachment(for busNumber: Int) -> NSTextAttachment {
+        let busIconTextSpacing: CGFloat = 5
+
+        // Instantiate busIconView off screen to later turn into UIImage
+        let busIconView = BusIcon(type: .blueBannerSmall, number: busNumber)
+
+        let busIconFrame = CGRect(
+            x: 0,
+            y: 0,
+            width: busIconView.intrinsicContentSize.width + busIconTextSpacing * 2,
+            height: busIconView.intrinsicContentSize.height
+        )
+
+        // Create container to add padding on sides
+        let containerView = UIView(frame: busIconFrame)
+        containerView.isOpaque = false
+        containerView.addSubview(busIconView)
+        busIconView.snp.makeConstraints { make in
+            make.leading.trailing.equalToSuperview().inset(busIconTextSpacing)
+            make.top.bottom.equalToSuperview()
+        }
+        view.addSubview(containerView)
```

Comment:
i think it makes most sense to first do `view.addSubview(containerView)` and then `containerView.addSubview(busIconView)` and set `busIconView`'s constraints